syntax="proto3";

package EnvironmentControl;

message StartExperimentRequest {
	string experiment_uuid = 1;
}

message StartExperimentReply {
	bool success = 1;
	string error_message = 2;
}

message ShutdownExperimentRequest{
	string experiment_name = 1;
	bool is_regex = 2;
}

message ShutdownExperimentReply{
	repeated string	experiment_names = 1;
}

message ListExperimentsRequest{

}

message ListExperimentsReply{
	repeated string	experiment_names = 1;
}

message GetQpidBrokerEndpointRequest{

}

message GetQpidBrokerEndpointReply{
	string endpoint = 1;
}

message GetTaoCosnamingEndpointRequest{

}

message GetTaoCosnamingEndpointReply{
	string endpoint = 1;
}